sdk/Javascript/src/Platforms/Platform.FxOS.js

```diff
@@ -0,0 +1,277 @@
+// ----------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// ----------------------------------------------------------------------------
+
+/// <reference path="..\Generated\MobileServices.DevIntellisense.js" />
+/*global $__fileVersion__:false, $__version__:false */
+
+var _ = require('Extensions');
+var Validate = require('Validate');
+var Promises = require('Promises');
```

Comment:
If there is a built in promise, and its A+ compliant, then yes we should see if it can be used.

If not, or its alot of work, we can continue to use this version.